### PR TITLE
Cleanup docs, add more type hints

### PR DESCRIPTION
**Pull request by GeorgeSittas**
Minor additions & fixes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
-It is a very comprehensive generic SQL parser with a robust [test suite](https://github.com/tobymao/sqlglot/blob/main/tests/). It is also quite [performant](#benchmarks) while being written purely in Python.
+It is a very comprehensive generic SQL parser with a robust [test suite](https://github.com/tobymao/sqlglot/blob/main/tests/). It is also quite [performant](#benchmarks), despite being written purely in Python.
 
 You can easily [customize](#custom-dialects) the parser, [analyze](#metadata) queries, traverse expression trees, and programmatically [build](#build-and-modify-sql) SQL.
 
-Syntax [errors](#parser-errors) are highlighted and dialect incompatibilities can warn or raise depending on configurations. However, it should be noted that the parser is very lenient when it comes to detecting errors, because it aims to consume as much SQL as possible. On one hand, this makes its implementation simpler, and thus more comprehensible, but on the other hand it means that syntax errors may sometimes go unnoticed.
```
Comment (GeorgeSittas) [inline README.md:9]: Thought this was lengthier than needed.

File: README.md [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+SQLGlot is a no-dependency SQL parser, transpiler, optimizer, and engine. It can be used to format SQL or translate between [19 different dialects](https://github.com/tobymao/sqlglot/blob/main/sqlglot/dialects/__init__.py) like [DuckDB](https://duckdb.org/), [Presto](https://prestodb.io/), [Spark](https://spark.apache.org/), [Snowflake](https://www.snowflake.com/en/), and [BigQuery](https://cloud.google.com/bigquery/). It aims to read a wide variety of SQL inputs and output syntactically correct SQL in the targeted dialects.
 
-It is a very comprehensive generic SQL parser with a robust [test suite](https://github.com/tobymao/sqlglot/blob/main/tests/). It is also quite [performant](#benchmarks) while being written purely in Python.
+It is a very comprehensive generic SQL parser with a robust [test suite](https://github.com/tobymao/sqlglot/blob/main/tests/). It is also quite [performant](#benchmarks), despite being written purely in Python.
```
Comment (tobymao) [inline README.md:5]: don't like the change here to despite

File: README.md (hunk 1 above)
Comment (tobymao) [reply README.md:9]: it should be explicit that it is not a goal to be a sql validator

**Review by tobymao — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (GeorgeSittas) [reply README.md:5]: What about "albeit"? Or should I just revert to "while"?

Comment (tobymao) [reply README.md:5]: just keep it

File: README.md (hunk 1 above)
Comment (GeorgeSittas) [reply README.md:9]: Tried to make it more explicit.

File: README.md (hunk 2 above)
Comment (GeorgeSittas) [reply README.md]: Done

File: README.md (hunk 1 above)
Comment (tobymao) [reply README.md:9]: it is not a goal at all to validate sql, we explicitly allow syntax errors like trailing commas

**Pull request merged**